backend/src/test/java/com/github/lenaruppert/backend/controller/MongoUserControllerTest.java

```diff
@@ -0,0 +1,190 @@
+                        .content("""
+                                {
+                                    "username": "test",
+                                    "password": "test"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().isOk());
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {
+                                    "username": "test",
+                                    "password": "test"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().isConflict());
+    }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "user", password = "password")
+    void loginUserWithValidUsernameAndPassword_Then200() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user/login")
+                .with(csrf())
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("""
+                        {
+                            "username": "user",
+                            "password": "password"
+                        }
+                        """));
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/user")
+                        .with(csrf()))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    @DirtiesContext
+    void loginUserWithInvalidUsernameAndPassword_Then401() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user/login")
+                        .with(csrf())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {
+                                    "username": "invalidUsername",
+                                    "password": "invalidPassword"
+                                }
+                                """))
+
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "user", password = "password")
+    void logoutUser_Then200() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user/logout")
+                        .with(csrf()))
+                .andExpect(status().isOk());
+
+    }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "user", password = "password")
+    void testGetMe2() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/user/me2")
+                        .with(csrf()))
+                .andExpect(status().isOk())
+                .andExpect(content().string("user"));
+
+    }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "user", password = "password")
+    void testGetAdminStatus() throws Exception {
```

Comment:
When I see this test, I think you might want to also test that a user without admin role doesn't have access to this page. But maybe you remove the endpoint anyways :)